openedx/openedx-learning
# Warn Django Admin users editing Collections [FC-0036]

## Description

https://github.com/openedx/openedx-learning/pull/229 added Django Admin screens for Collections, but testing [revealed issues](https://github.com/openedx/edx-platform/pull/35496#issuecomment-2369978216) if Collections are edited in the LMS, due to missed signal handlers in the Studio Search app.

This PR adds warnings shown when editing a Collection, to encourage users to use the Studio Django Admin instead of the LMS. Also removes the configuration that allowed editing Collections from the Django Admin list view (so the warnings on the edit screen will be seen), and improves the "help" text for the "enabled" checkbox.

![edit_only_in_studio](https://github.com/user-attachments/assets/8fddb2cd-9ebd-4f45-8a12-9ff0517b12aa)

## Supporting information

Related to: https://github.com/openedx/modular-learning/issues/231
Private-ref: [FAL-3809](https://tasks.opencraft.com/browse/FAL-3809)

## Testing instructions

1. Using the API, create a couple collections (`python manage.py shell`):
   ```python
   from openedx_learning.authoring import api
   learning_package = api.create_learning_package("FAL-3809", "Learning package for FAL-3809")
   api.create_collection(learning_package.id, "FAL-3809-1", title="Collection 1 for FAL-3809")
   api.create_collection(learning_package.id, "FAL-3809-2", title="Collection 2 for FAL-3809")
   ```
1. View the collection in the Django Admin: (`python manage.py runserver`), and ensure you can:
   * edit the collections' title and/or description.
   * Note the warning shown.

## Deadline

ASAP
